packages/app/src/specs/generators/EmptyGenerator.vue

```diff
@@ -190,6 +195,24 @@ const hasError = computed(() => !isValidSpecFile.value && !!specFile.value)
 
 const result = ref<GeneratorSuccessFileFragment | null>(null)
 
+onMounted(() => {
+  // Focus text input and try to pre-select file name
+  inputRef.value?.focus()
+
+  // This regex matches a group of word characters, spaces, and '-' characters before a '.' character.
+  // For example 'path/to/spec/spec-name.cy.tsx' ---> 'spec-name'
+  const fileNameRegex = /[ \w-]+(?=\.)/
+  const inputValue = inputRef.value?.value
+  const match = inputValue?.match(fileNameRegex)
+
+  if (inputValue && match) {
+    const startSelectionIndex = match.index || 0
+    const endSelectionIndex = inputValue.indexOf(match[0]) + match[0].length
```

Comment:
I think you're still doing a bit of extra work here - couldn't this `inputValue.indexOf(match[0])` be replaced by `startSelectionIndex`?

Reply:
yeah you're right, it could be 👍🏻